ckanext/harvest/logic/action/update.py

```diff
@@ -518,9 +560,13 @@ def harvest_source_reindex(context, data_dict):
     if package_dict.get('config'):
         config = json.loads(package_dict['config'])
         for key, value in package_dict.iteritems():
-            if key not in config:
-                new_dict[key] = value
+            if value:
+                if value and key not in config:
```

Comment:
the duplicated `if value` is my mistake (shouldn't rush changes, lesson learnt). The `if value` fixed the issue of deleting harvest endpoints which had a config (my comment mentioned here: https://github.com/ckan/ckanext-harvest/commit/58a873ac7a31e7bfc956967d4e6f49368e3799e7#commitcomment-14129955). However I see now that, what I thought was a good fix, could break other things. I need to look for a solution to the `revision_timestamp` and `tracking_summary` values having missing objects.